atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraClientPool.java

```diff
@@ -432,6 +433,17 @@ public InetSocketAddress apply(String endpoint) {
             } catch (Exception e) {
                 numTries++;
                 this.<K>handleException(numTries, hostPool.getHost(), e);
+                if (isRetriableWithBackoffException(e)) {
+                    log.warn("Retrying with backoff a query intended for host {}.", hostPool.getHost(), e);
+                    try {
+                        Thread.sleep(numTries * 1000);
+                    } catch (InterruptedException i) {
+                        Thread.currentThread().interrupt();
```

Comment:
not sure if this will do what you intend, which is to let this call get interrupted and stop. I don't think anything else in this call will actually check the interrupted flag and stop. maybe just rethrow unchecked?